Source/Scene/ModelExperimental/ModelExperimental.js

```diff
@@ -0,0 +1,254 @@
+ * If this object was destroyed, it should not be used; calling any function other than
+ * <code>isDestroyed</code> will result in a {@link DeveloperError} exception.
+ *
+ * @returns {Boolean} <code>true</code> if this object was destroyed; otherwise, <code>false</code>.
+ *
+ * @see ModelExperimental#destroy
+ */
+ModelExperimental.prototype.isDestroyed = function () {
+  return false;
+};
+
+/**
+ * Destroys the WebGL resources held by this object.  Destroying an object allows for deterministic
+ * release of WebGL resources, instead of relying on the garbage collector to destroy this object.
+ * <br /><br />
+ * Once an object is destroyed, it should not be used; calling any function other than
+ * <code>isDestroyed</code> will result in a {@link DeveloperError} exception.  Therefore,
+ * assign the return value (<code>undefined</code>) to the object as done in the example.
+ *
+ * @exception {DeveloperError} This object was destroyed, i.e., destroy() was called.
+ *
+ *
+ * @example
+ * model = model && model.destroy();
+ *
+ * @see ModelExperimental#isDestroyed
+ */
+ModelExperimental.prototype.destroy = function () {
+  var gltfLoader = this._gltfLoader;
+  if (defined(gltfLoader)) {
+    gltfLoader.destroy();
+  }
+  destroyObject(this);
+};
+
+function initialize(model, options) {
+  var gltf = options.gltf;
+
+  var loaderOptions = {
+    baseResource: options.basePath,
+    releaseGltfJson: options.releaseGltfJson,
+    incrementallyLoadTextures: options.incrementallyLoadTextures,
+  };
+
+  if (gltf instanceof Uint8Array) {
+    loaderOptions.typedArray = gltf;
+    loaderOptions.gltfResource = Resource.createIfNeeded(
+      defaultValue(options.basePath, "")
+    );
+  } else {
+    loaderOptions.gltfResource = gltf;
+  }
+
+  var loader = new GltfLoader(loaderOptions);
+  model._gltfLoader = loader;
+  loader.load();
+
+  loader.promise
+    .then(function (loader) {
+      model._sceneGraph = new ModelExperimentalSceneGraph({
+        model: model,
+        modelComponents: loader._components,
+        upAxis: options.upAxis,
+        forwardAxis: options.forwardAxis,
+        modelMatrix: options.modelMatrix,
+      });
+      model._resourcesLoaded = true;
+      model._ready = true;
+      model._readyPromise.resolve(model);
+    })
+    .otherwise(function () {
+      ModelExperimentalUtility.getFailedLoadFunction(
+        this,
+        "model",
+        options.basePath
+      );
+    });
+}
+
+/**
+ *
+ * @param {Object}} options Object with the following properties:
+ * @param {Resource|String} options.url The url to the .gltf or .glb file.
+ * @param {Object} [options.basePath=''] The base path that paths in the glTF JSON are relative to.
+ * @param {Matrix4} [options.modelMatrix=Matrix4.IDENTITY] The 4x4 transformation matrix that transforms the model from model to world coordinates.
+ * @param {Boolean} [options.incrementallyLoadTextures] Determine if textures may continue to stream in after the model is loaded.
```

Comment:
Make sure these parameter lists are consistent between `fromGltf` and the constructor (where applicable)